frontend/model/contracts/chatroom.js

```diff
@@ -520,6 +521,105 @@ sbp('chelonia/defineContract', {
       sideEffect ({ contractID, hash }) {
         emitMessageEvent({ contractID, hash })
       }
+    },
+    'gi.contracts/chatroom/voteOnPoll': {
+      validate: objectOf({
+        hash: string,
+        votes: arrayOf(string)
+      }),
+      process ({ data, meta, hash, id }, { state }) {
+        const msgIndex = findMessageIdx(data.hash, state.messages)
+
+        if (msgIndex >= 0) {
```

Comment:
@SebinSong, this line is the reason of the error I reported.
`state.messages` could not have the message with that `hash`. Because `state.messages` doesn't save all the messages created from the beginning. I remember you couldn't see any messages in vuex contract state.
Chatroom contract creates messages only when `onlyRenderMessages` attribute is `true`, only inside Chat page.

Reply:
Even though `msgIndex === -1`, `NOTIFICATION` message should be added.

```
if (msgIndex >= 0) {
  ...
  Vue.set(state.messages[msgIndex], 'pollData', { ...pollData, options: optsCopy })
}
...
state.messages.push(newMessage)
```

Reply:
@Silver-IT 
updated per your comment. Btw,

> Chatroom contract creates messages only when onlyRenderMessages attribute is true, only inside Chat page.

Where exactly is this happening in the `GroupChat.vue`? I've been searching the source-code trying to locate the part where this takes place but couldn't easily spot it.

Reply:
Since there could be really many messages in the chatroom contract, we couldn't save those messages inside contract. So we creates messages only when `onlyRenderMessages` attribute is set as true.

Reply:
@SebinSong 
> Where exactly is this happening in the `GroupChat.vue`?

`onlyRenderMessage` is set in `ChatMain.vue`